etcd/etcd.go

```diff
@@ -204,13 +203,13 @@ func (c *Client) tryEndpoints(mreq func(string) (*http.Request, error)) (*http.R
 		}
 
 		endpointErrs = append(endpointErrs, err)
-		endpoints = endpoints[1:]
 	}
 
-	rotate := len(c.endpoints) - len(endpoints)
-	c.endpoints = append(c.endpoints[rotate:], c.endpoints[:rotate]...)
+	if lastEndpoint > 0 {
+		if lastEndpoint > 1 {
+			c.endpoints = append(c.endpoints[lastEndpoint:], c.endpoints[:lastEndpoint]...)
```

Comment:
if lastEndpoint <= 1 this means that there is no checkpoints or only one checkpoint. Rotation haven't sense.